test/gen_dune_rules.ml

```diff
@@ -0,0 +1,30 @@
+let is_testcase name =
+  Fpath.(has_ext "dot" (v name))
+
+let pp_targets ppf filenames =
+  Fmt.pf ppf
+    {|(rule
+ (targets %a)
+ (action (run ./test.exe)))
+
+|}
+    Fmt.(list ~sep:(const string "\n          ") string) filenames
+
+let pp_runtest ppf filename =
+  Fmt.pf ppf
+    {|(alias
+ (name runtest)
+ (package current)
+ (action (diff expected/%s %s)))
+|}
+    filename filename
+
+let pp_duneinc = Fmt.(pp_targets ++ (list ~sep:Format.pp_force_newline pp_runtest))
+
+let () =
+  Sys.readdir "expected"
+  |> Array.to_list
+  |> List.sort String.compare
+  |> List.filter is_testcase
+  |> pp_duneinc Fmt.stdout
```

Comment:
I suspect you should be using something like `Fmt.pr "@[<v>%a@]@."` here to flush the output at the end.